addons/binding/org.openhab.binding.opendaikin/src/main/java/org/openhab/binding/opendaikin/internal/handler/OpenDaikinAcUnitHandler.java

```diff
@@ -0,0 +1,267 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing OpenDaikin AC Unit");
+        OpenDaikinConfiguration config = getConfigAs(OpenDaikinConfiguration.class);
+        if (config.host == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Host address must be set");
+        } else {
+            webTargets = new OpenDaikinWebTargets(client, config.host);
+            refreshInterval = config.refresh;
+
+            schedulePoll();
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        super.handleRemoval();
+        stopPoll();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private void schedulePoll() {
+        if (pollFuture != null) {
+            pollFuture.cancel(false);
+        }
+        logger.debug("Scheduling poll for 500ms out, then every {} ms", refreshInterval);
+        pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 500, refreshInterval, TimeUnit.MILLISECONDS);
+    }
+
+    private synchronized void stopPoll() {
+        if (pollFuture != null && !pollFuture.isCancelled()) {
+            pollFuture.cancel(true);
+            pollFuture = null;
+        }
+    }
+
+    private synchronized void poll() {
+        try {
+            logger.debug("Polling for state");
+            pollStatus();
+        } catch (IOException e) {
+            logger.debug("Could not connect to Daikin controller", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (RuntimeException e) {
+            logger.warn("Unexpected error connecting to Daikin controller", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void pollStatus() throws IOException {
+        ControlInfo controlInfo = webTargets.getControlInfo();
+        updateStatus(ThingStatus.ONLINE);
+        if (controlInfo != null) {
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_POWER, controlInfo.power ? OnOffType.ON : OnOffType.OFF);
+            if (controlInfo.temp.isPresent()) {
+                updateTempState(OpenDaikinBindingConstants.CHANNEL_AC_TEMP, controlInfo.temp.get());
+            }
+
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_MODE, new StringType(controlInfo.mode.name()));
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_FAN_SPEED, new StringType(controlInfo.fanSpeed.name()));
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_FAN_DIR, new StringType(controlInfo.fanMovement.name()));
+        }
+
+        SensorInfo sensorInfo = webTargets.getSensorInfo();
+        if (sensorInfo != null) {
+            if (sensorInfo.indoortemp.isPresent()) {
```

Comment:
This logic could imho easily be handled by `updateTemperatureChannel ` (suggested rename ;-))

```java
updateTemperatureChannel (OpenDaikinBindingConstants.CHANNEL_INDOOR_TEMP, sensorInfo.getIndoorTemperature());

...

private void updateTemperatureChannel(Optional<Double?> maybeTemperature) {
    updateState(channel, maybeTemperature
        .map(t -> new QuantityType<Temperature>(new DecimalType(t), CELSIUS)
        .orElse(UnDefType.UNDEF);

```
@Hilbrand should this be `SIUnits.CELSIUS`? or something with SmartHome?

Reply:
> @Hilbrand should this be SIUnits.CELSIUS? or something with SmartHome?